src/compiler/binder.ts

```diff
@@ -135,6 +135,26 @@ module ts {
             return node.name ? declarationNameToString(node.name) : getDeclarationName(node);
         }
 
+        /* internal */
+        /**
+         * Checks if the symbol contains a class declaration declaration that is non-ambient.
+         */
+        function hasNonAmbientClass(symbol: Symbol): boolean {
+            if (symbol) {
+                return !! forEach(symbol.declarations, (element: Declaration, index: number) => {

```

Comment:
If I don't include `!!`, the object returned by `forEach` is truthy/falsy (and when treated as a `boolean`, its value is unchanged), but !! makes the actual value returned `true` or `false`. Does it matter from a performance/runtime safety point of view? If not, I'll remove the `!!`.

Also the style guidelines suggest using `ts.forEach` instead of for loops, so barring a strong reason to change, I'll keep it as is.